app/src/test/java/com/example/forzenbook/domain/DomainUnitTest.kt

```diff
@@ -0,0 +1,83 @@
+import com.hamdan.forzenbook.data.repository.CreateAccountRepository
+import com.hamdan.forzenbook.data.repository.LoginRepository
+import com.hamdan.forzenbook.domain.usecase.login.CreateAccountRequestUseCaseImpl
+import com.hamdan.forzenbook.domain.usecase.login.CreateAccountResult
+import com.hamdan.forzenbook.domain.usecase.login.LoginGetTokenUseCaseImpl
+import com.hamdan.forzenbook.domain.usecase.login.LoginRequestValidationUseCaseImpl
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert
+import org.junit.Assert.fail
+import org.junit.Test
+
+class DomainUnitTest {
+    @Test
+    fun `CreateAccountRequestUseCaseImpl test, checks all branches`() {
+        val succeedRepo = mockk<CreateAccountRepository>()
+        val accountExistsRepo = mockk<CreateAccountRepository>()
+        val failureRepo = mockk<CreateAccountRepository>()
+
+        coEvery { succeedRepo.createUser("", "", "", "", "") } returns Unit
+        coEvery { accountExistsRepo.createUser("", "", "", "", "") } throws AccountException("")
+        coEvery { failureRepo.createUser("", "", "", "", "") } throws Exception("")
+
+        var useCase = CreateAccountRequestUseCaseImpl(succeedRepo)
+        runBlocking {
+            Assert.assertEquals(useCase("", "", "", "", ""), CreateAccountResult.CREATE_SUCCESS)
+        }
+        useCase = CreateAccountRequestUseCaseImpl(accountExistsRepo)
+        runBlocking {
+            Assert.assertEquals(useCase("", "", "", "", ""), CreateAccountResult.CREATE_EXISTS)
+        }
+        useCase = CreateAccountRequestUseCaseImpl(failureRepo)
+        runBlocking {
+            Assert.assertEquals(useCase("", "", "", "", ""), CreateAccountResult.CREATE_FAILURE)
+        }
+    }
+
+    @Test
+    fun `RequestValidation Usecase test, checks that the invocation is successful`() {
+        val successRepo = mockk<LoginRepository>()
+        coEvery { successRepo.requestValidation("") } returns Unit
+        val useCase = LoginRequestValidationUseCaseImpl(successRepo)
+        runBlocking {
+            try {
+                useCase("")
+            } catch (e: Exception) {
+                fail()
+            }
+        }
+    }
+
+    @Test
+    fun `LoginGetToken Usecase test, checks success and whether it properly throws an Exception`() {
+        val successRepo = mockk<LoginRepository>()
+        val nullToken = mockk<LoginRepository>()
+        val testString = "someString"
+        coEvery { successRepo.getToken("", "") } returns LoginResponse(testString)
+        coEvery { nullToken.getToken("", "") } returns LoginResponse(null)
+
+        var useCase = LoginGetTokenUseCaseImpl(successRepo)
+        runBlocking {
+            try {
+                useCase("", "")
+            } catch (e: Exception) {
+                fail()
+            }
+        }
+        useCase = LoginGetTokenUseCaseImpl(nullToken)
+        runBlocking {
```

Comment:
The repo can fail to get from DB, make sure that works
The internet can be not connected, make sure that works
Where are any tests about the validation of each of the create account fields?

Reply:
Once FA-84 is accepted I can make the test for the DB branch cases, currently the DB branch cases are not in the code so I can't really test them here.
What kind of validation am I makeing on the create account fields? In the code the only places they would be getting validated is when they are being put in and when the server probably sees them, so what validity checks would I be doing here in the usecase layer?